File: GameLibrary/test/cc/lib/reflector/OverridesSerialize.java
package cc.lib.reflector;

import java.io.IOException;

public class OverridesSerialize extends Reflector<OverridesSerialize> {

    String msg;
    int x;
    
    public OverridesSerialize() {}
    
    public OverridesSerialize(String msg, int x) {
        super();
        this.msg = msg;
        this.x = x;
    }

    @Override
    public void serialize(RPrintWriter out) throws IOException {
        out.println(msg + " " + x);
    }

    @Override
    public void deserialize(RBufferedReader input) throws IOException {
        String text = input.readLine().trim();
        String[] parts = text.split("[ ]+");
        msg = parts[0];
        x = Integer.parseInt(parts[1]);
    }
    
    public boolean equals(Object o) {
    	if (o == null || !(o instanceof OverridesSerialize))
    		return false;
    	
    	OverridesSerialize os = (OverridesSerialize)o;
    	return msg.equals(os.msg) && x == os.x;
    }
}

File: GameLibrary/test/cc/lib/kreflector/OverridesSerialize.kt
package cc.lib.kreflector

import java.io.IOException

class OverridesSerialize : KReflector<OverridesSerialize> {
	var msg: String? = null
	var x = 0

	constructor() {}
	constructor(msg: String?, x: Int) : super() {
		this.msg = msg
		this.x = x
	}

	@Throws(IOException::class)
	public override fun serialize(out: RPrintWriter) {
		out.println("$msg $x")
	}

	@Throws(IOException::class)
	public override fun deserialize(reader: RBufferedReader) {
		reader.readLine()?.trim { it <= ' ' }?.let { text ->
			val parts = text.split("[ ]+".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
			msg = parts[0]
			x = parts[1].toInt()
		}
	}

	override fun equals(o: Any?): Boolean {
		if (o == null || o !is OverridesSerialize) return false
		val os = o
		return msg == os.msg && x == os.x
	}
}